### PR TITLE
util: Add vk_struct_helper.hpp

### LINKED ISSUE #74: vk_typemap_helper.h
Both VVL and VEL have the same helper file:

https://github.com/KhronosGroup/Vulkan-ValidationLayers/blob/main/layers/vulkan/generated/vk_typemap_helper.h
https://github.com/KhronosGroup/Vulkan-ExtensionLayer/blob/main/utils/generated/vk_typemap_helper.h

### PR DESCRIPTION
**Pull request by charles-lunarg**
Move the vk_struct_helper.h file in Vulkan-ValidationLayers over to this repo.

Changes from that include:
 * Renaming functions
 * Placing them in the `vku` namespace
 * Adding the `InitStructHelper` class which deduces the type based on the variable that is being initialized
  * If a type isn't defined then compiler error is emitted
  
 Resolves #74

### CODE REVIEW

File: include/vulkan/utility/vk_typemap_helper.hpp [hunk 1]
```diff
@@ -0,0 +1,1890 @@
...
+}
+
+// Find an entry of the given type in the const pNext chain
+template <typename T> const T *LvlFindInChain(const void *next) { return vku::FindStructInPNextChain<T>(next); }
+
+// Find an entry of the given type in the pNext chain
+template <typename T> T *LvlFindModInChain(void *next) { return vku::FindStructInPNextChain<T>(next); }
+
+// Find last element of pNext chain
+inline VkBaseOutStructure *LvlFindLastInChain(void *next) { return vku::FindLastStructInPNextChain(next); }
```
Comment (juan-lunarg) [inline include/vulkan/utility/vk_typemap_helper.hpp:1888]: 👎🏾 

If we don't fix this now when will we fix it? I understand it will be a pain to fix.

**Review by juan-lunarg — CHANGES_REQUESTED** [review]

File: include/vulkan/utility/vk_typemap_helper.hpp (hunk 1 above)
Comment (charles-lunarg) [reply include/vulkan/utility/vk_typemap_helper.hpp:1888]: Thats fair, why have long term pain when some short term will lessen it.

**Comment by charles-lunarg** [comment]
`vk_typemap_helper.hpp` is not a great name, since there isn't a 'typemap' in the code anywhere. `vk_struct_init.hpp` is better but I'm not sold.

**Comment by juan-lunarg** [comment]
What about `vk_struct_helper.hpp`? Since this also does pNext searching?

**Comment by charles-lunarg** [comment]
Thats a good idea due to the inclusion of pNext chain helpers as well.

**Review by juan-lunarg — COMMENTED** [review]
Looks good overall aside from chromium build issue.

**Comment by charles-lunarg** [comment]
build.gn has been updated.

**Review by juan-lunarg — APPROVED** [review]
Update PR title with correct header file

**Comment by juan-lunarg** [comment]
FWIW I'd start with updating the extension layer instead of vvl.

**Pull request merged**